presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkDictionaryWriter.java

```diff
@@ -0,0 +1,303 @@
+        if (type.equals(VARCHAR)) {
+            columnWriter = new SliceDirectColumnWriter(COLUMN_INDEX, type, compressionParameters, Optional.empty(), DWRF, this::newStringStatisticsBuilder, DWRF.createMetadataWriter());
+        }
+        else {
+            columnWriter = new LongColumnWriter(COLUMN_INDEX, type, compressionParameters, Optional.empty(), DWRF, IntegerStatisticsBuilder::new, DWRF.createMetadataWriter());
+        }
+        for (Block block : data.getBlocks()) {
+            columnWriter.beginRowGroup();
+            columnWriter.writeBlock(block);
+            columnWriter.finishRowGroup();
+        }
+        columnWriter.close();
+        columnWriter.reset();
+    }
+
+    @Benchmark
+    public void writeDictionary(BenchmarkData data)
+    {
+        ColumnWriter columnWriter;
+        Type type = data.getType();
+        if (type.equals(VARCHAR)) {
+            columnWriter = new SliceDictionaryColumnWriter(COLUMN_INDEX, type, compressionParameters, Optional.empty(), DWRF, DEFAULT_MAX_STRING_STATISTICS_LIMIT, DWRF.createMetadataWriter());
+        }
+        else {
+            columnWriter = new LongDictionaryColumnWriter(COLUMN_INDEX, type, compressionParameters, Optional.empty(), DWRF, DWRF.createMetadataWriter());
+        }
+        for (Block block : data.getBlocks()) {
+            columnWriter.beginRowGroup();
+            columnWriter.writeBlock(block);
+            columnWriter.finishRowGroup();
+        }
+        columnWriter.close();
+        columnWriter.reset();
+    }
+
+    @Benchmark
+    public void writeDictionaryAndConvert(BenchmarkData data)
+    {
+        DictionaryColumnWriter columnWriter;
+        Type type = data.getType();
+        if (type.equals(VARCHAR)) {
+            columnWriter = new SliceDictionaryColumnWriter(COLUMN_INDEX, type, compressionParameters, Optional.empty(), DWRF, DEFAULT_MAX_STRING_STATISTICS_LIMIT, DWRF.createMetadataWriter());
+        }
+        else {
+            columnWriter = new LongDictionaryColumnWriter(COLUMN_INDEX, type, compressionParameters, Optional.empty(), DWRF, DWRF.createMetadataWriter());
+        }
+        for (Block block : data.getBlocks()) {
+            columnWriter.beginRowGroup();
+            columnWriter.writeBlock(block);
+            columnWriter.finishRowGroup();
+        }
+        int maxDirectBytes = toIntExact(new DataSize(128, MEGABYTE).toBytes());
+        columnWriter.tryConvertToDirect(maxDirectBytes);
+        columnWriter.close();
+        columnWriter.reset();
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private static final int NUM_BLOCKS = 10_000;
+        private static final int ROWS_PER_BLOCK = 1_000;
+        private final Random random = new Random(0);
+        private final List<Block> blocks;
+        @Param({
+                "integer",
+                "bigint",
+                "varchar"
+        })
+        private String typeSignature = "integer";
```

Comment:
final

Reply:
They are benchmark params and benchmark sets them after the object is created via reflection. So they can't be final.